firmware/src/ui/text.cpp

```
@@ -0,0 +1,174 @@
+    put(note, kAltNote[static_cast<int>(track.alt)]);
+  }
+}
+
+// Starts a line: the pad's name padded to the notation column, or spaces for a
+// continuation. False when the screen is full.
+bool begin_line(TextLines& out, int track, const char* name, int max_lines) {
+  if (out.count >= max_lines) return false;
+  char* line = out.text[out.count];
+  std::memset(line, ' ', kTextNameColumns);
+  line[kTextNameColumns] = '\0';
+  if (name != nullptr) std::memcpy(line, name, std::strlen(name));
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)